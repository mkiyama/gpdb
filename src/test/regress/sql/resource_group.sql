-- ----------------------------------------------------------------------
-- Test: assign/alter a role to a resource group
-- ----------------------------------------------------------------------

--start_ignore
DROP ROLE IF EXISTS rg_test_role;
--end_ignore
CREATE ROLE rg_test_role;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
ALTER ROLE rg_test_role RESOURCE GROUP non_exist_group;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
ALTER ROLE rg_test_role RESOURCE GROUP admin_group;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
ALTER ROLE rg_test_role RESOURCE GROUP none;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';

DROP ROLE rg_test_role;
CREATE ROLE rg_test_role SUPERUSER;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
ALTER ROLE rg_test_role RESOURCE GROUP default_group;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
ALTER ROLE rg_test_role RESOURCE GROUP none;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';

DROP ROLE rg_test_role;
CREATE ROLE rg_test_role RESOURCE GROUP non_exist_group;

-- nonsuper user should not be assigned to admin group
CREATE ROLE rg_test_role RESOURCE GROUP admin_group;
