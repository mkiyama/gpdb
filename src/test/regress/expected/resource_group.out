-- ----------------------------------------------------------------------
-- Test: assign/alter a role to a resource group
-- ----------------------------------------------------------------------
--start_ignore
DROP ROLE IF EXISTS rg_test_role;
NOTICE:  role "rg_test_role" does not exist, skipping
--end_ignore
CREATE ROLE rg_test_role;
NOTICE:  resource queue required -- using default resource queue "pg_default"
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
 rolresgroup 
-------------
        6437
(1 row)

ALTER ROLE rg_test_role RESOURCE GROUP non_exist_group;
ERROR:  resource group "non_exist_group" does not exist
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
 rolresgroup 
-------------
        6437
(1 row)

ALTER ROLE rg_test_role RESOURCE GROUP admin_group;
ERROR:  only super user can be assigned to admin resgroup
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
 rolresgroup 
-------------
        6437
(1 row)

ALTER ROLE rg_test_role RESOURCE GROUP none;
NOTICE:  resource group required -- using default resource group "default_group"
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
 rolresgroup 
-------------
        6437
(1 row)

DROP ROLE rg_test_role;
CREATE ROLE rg_test_role SUPERUSER;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
 rolresgroup 
-------------
        6438
(1 row)

ALTER ROLE rg_test_role RESOURCE GROUP default_group;
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
 rolresgroup 
-------------
        6437
(1 row)

ALTER ROLE rg_test_role RESOURCE GROUP none;
NOTICE:  resource group required -- using default resource group "admin_group"
SELECT rolresgroup FROM pg_authid WHERE rolname = 'rg_test_role';
 rolresgroup 
-------------
        6438
(1 row)

DROP ROLE rg_test_role;
CREATE ROLE rg_test_role RESOURCE GROUP non_exist_group;
NOTICE:  resource queue required -- using default resource queue "pg_default"
ERROR:  resource group "non_exist_group" does not exist
-- nonsuper user should not be assigned to admin group
CREATE ROLE rg_test_role RESOURCE GROUP admin_group;
NOTICE:  resource queue required -- using default resource queue "pg_default"
ERROR:  only super user can be assigned to admin resgroup
